File: test/fixtures/small-swift/shape.swift
/// A drawable shape.
public protocol Shape {
    /// Returns the area of the shape.
    func area() -> Double
}

/// A circle with a radius.
public struct Circle: Shape {
    public let radius: Double

    public func area() -> Double {
        return radius * radius * 3.14
    }
}

/// Kinds of shapes.
public enum Kind {
    case round
    case pointy(Int)
}

/// Adds a label to circles.
extension Circle {
    /// A human-readable label.
    func label() -> String {
        return describe()
    }

    private func describe() -> String {
        return "circle"
    }
}

/// Builds the default unit circle.
public func defaultCircle() -> Circle {
    return Circle(radius: 1.0)
}

File: test/fixtures/small-rust/shape.rs
/// A drawable shape.
pub trait Shape {
    /// Returns the area of the shape.
    fn area(&self) -> f64;

    /// Human-readable name, with a default.
    fn name(&self) -> &str {
        "shape"
    }
}

/// A circle with a radius.
pub struct Circle {
    pub radius: f64,
}

impl Shape for Circle {
    fn area(&self) -> f64 {
        self.radius * self.radius * 3.14
    }
}

/// Kinds of shapes.
pub enum Kind {
    Round,
    Pointy(u32),
}

/// Builds the default unit circle.
pub fn default_circle() -> Circle {
    Circle { radius: 1.0 }
}
